addons/account/wizard/setup_wizards.py

```diff
@@ -121,11 +121,12 @@ def _onchange_new_journal_related_data(self):
 
     @api.depends('journal_id')  # Despite its name, journal_id is actually a One2many field
     def _compute_linked_journal_id(self):
+        journal_type = self.env.context.get('journal_type', 'bank')
         for record in self:
-            record.linked_journal_id = record.journal_id and record.journal_id[0] or record.default_linked_journal_id()
+            record.linked_journal_id = record.journal_id and record.journal_id[0] or record.default_linked_journal_id(journal_type)
 
-    def default_linked_journal_id(self):
-        for journal_id in self.env['account.journal'].search([('type', '=', 'bank'), ('bank_account_id', '=', False)]):
+    def default_linked_journal_id(self, journal_type):
+        for journal_id in self.env['account.journal'].search([('type', '=', journal_type), ('bank_account_id', '=', False)]):
             empty_journal_count = self.env['account.move'].search_count([('journal_id', '=', journal_id.id)])
```

Comment:
May I  ask you to clean that function and optimize it. 
It seems I miss that `search_count` in the loop :sweat_smile: 
```suggestion
        journals_with_moves = self.env['account.move'].search_fetch(
            [
                ('journal_id', '!=', False),
                ('journal_id.type', '=', journal_type),
            ],
            ['journal_id'],
        ).journal_id
        return self.env['account.journal'].search(
            [
                ('type', '=', journal_type),
                ('bank_account_id', '=', False),
                ('id', 'not in', journals_with_moves.ids)
            ],
            limit=1,
        )
```
I think this change should do the job